doc/testing.md

```diff
@@ -638,6 +640,29 @@ double click on it; enable or disable hotkeys.
 
 Note: restart is required to make the settings take effect.
 
+#### Robot API
+
+Most part of automated Client UI tests use Robot API to control the UI. Usually
+the default operating system settings need to be adjusted for correct work of
+Robot functionality. The detailed steps how to access and update that settings
+for different platforms are provided below.
+
+##### MacOS
+
+Robot functionality is not permitted to control your Mac by default starting
+from MacOS 10.15. To enable it choose Apple menu; System Settings, click
+Privacy & Security; select Accessibility and ensure the following apps are
+allowed to control your computer: `Java` and `Terminal`. If the tests are run
+from an IDE, the IDE should be granted this permission too.
+
+##### Windows
+
+On Windows there is a delay in focus transfer. Usually it causes automated
+UI test failure. To discard the delay type `regedit` in the Search and then
+select Registry Editor; navigate to the following key:
+`HKEY_CURRENT_USER\Control Panel\Desktop\ForegroundLockTimeout`; make sure
+its value is set to 0.
```

Comment:
Probably it should mention that this is a workaround for the usage of Cygwin:
https://bugs.openjdk.org/browse/CODETOOLS-7901623

Reply:
Done